### ISSUE TITLE
Release 4.3.0 - Manual tests - Active response

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#12870]
The following issue aims to run the specified test for the current release candidate, report the results, and open new issues for any encountered errors.

## Test information
|                       |                                            |
|-----------------------|--------------------------------------------|
| **Test name**         |  Active response                                          |
| **Category**          | Active response                                           |
| **Deployment option** | OVA                                           |
| **Main release issue** | https://github.com/wazuh/wazuh/issues/10954                                           |
| **Release candidate #** | RC5                                           |

## Test description

Test that the following proof of concept cases work in 4.3.0:
- https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html
- https://documentation.wazuh.com/current/proof-of-concept-guide/detect-remove-malware-virustotal.html
- https://documentation.wazuh.com/current/proof-of-concept-guide/detect-malware-yara-integration.html

Test that this blog post still works well with 4.3.0:
- https://wazuh.com/blog/how-to-integrate-yara-with-wazuh

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                            |
|---------------------------------|--------------------------------------------|
| :green_circle:  | All checks passed. |
| :red_circle:  | There is at least one failed check. |
| :yellow_circle:  | There is at least one expected fail or skipped test and no failures. |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. 

An extended report of the test results must be attached as a zip or txt. Please attach any documents, screenshots or tables to the issue update with the results. This report can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

After the review of the proof of concept and use cases done in this issue's comments, the following issues were opened and should be reviewed and taken into account for this Wazuh release.

|                |             |                     |                |
|----------------|-------------|---------------------|----------------|
| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|:red_circle:|[Blocking a malicious actor](https://github.com/wazuh/wazuh/issues/12870#issuecomment-1081716546)|Proof of Concept|Opened issue https://github.com/wazuh/wazuh/issues/12919 <br /> Opened issue https://github.com/wazuh/wazuh-documentation/issues/4979|
|:red_circle:|[Detecting and removing malware using the VirusTotal Integration](https://github.com/wazuh/wazuh/issues/12870#issuecomment-1082002324)|Proof of Concept|Opened issue https://github.com/wazuh/wazuh-documentation/issues/4981|
|:red_circle:|[How to integrate Wazuh with YARA](https://github.com/wazuh/wazuh/issues/12870#issuecomment-1084643157)|Use case|Opened issue https://github.com/wazuh/wazuh-documentation/issues/5000|
|:red_circle:|[Detecting malware using Yara integration](https://github.com/wazuh/wazuh/issues/12870#issuecomment-1084760684)|Proof of Concept|Opened issue https://github.com/wazuh/wazuh/issues/12976 <br /> Opened issue https://github.com/wazuh/wazuh-documentation/issues/5001|
|:red_circle:|Installing a Wazuh agent from sources in a CentOS8 endpoint|Documentation|Opened issue https://github.com/wazuh/wazuh-documentation/issues/4984|

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @chemamartinez  
- [x] @davidjiglesias

**Comment by mcarmona99** [comment]
# Active response manual tests

Wazuh branch: `4.3.0-rc5`
Wazuh documentation branch: `4.3-rc`

## Installation 

### OVA deployment option 🟢 

Everything is OK, no errors were found.

**Comment by mcarmona99** [comment]
## Active response tests

### Proof of Concept guide > Blocking a malicious actor 🔴 

#### Prerequisites

1.- Apache server running on CentOS8 (the OVA is using a CentOS 7 version, so I will use CentOS 7 (CentOS Linux release 7.9.2009 (Core)). Install Apache server:

As Apache is an external tool, I understand that no more indications are given here. In spite of it, I think some guides or links could be OK. Ref: https://httpd.apache.org/

```
sudo yum update httpd
sudo yum install httpd
```

Start and enable the Apache web server.


2.- Configure the Wazuh CentOS 8 host to monitor the Apache access logs in the `/var/ossec/etc/ossec.conf` configuration file. Added `localfile` configuration to monitor apache logs.

```
<localfile>
<log_format>apache</log_format>
<location>/var/log/httpd/access_log</location>
</localfile>
```

#### Configuration

1.- Added the Alienvault IP reputation DB to the Wazuh manager -> `/var/ossec/etc/lists/alienvault_reputation.ipset`

2.- Add the IP address of the Windows endpoint to the Alienvault IPs list. To do this, I have downloaded a Windows 7 OVA.

![image](https://user-images.githubusercontent.com/57481331/160593603-de6400cf-2023-43b3-9ee2-851a7b61ddd7.png)


3.- Download script to convert from `ipset` format to `cdb list` format.

4.- Convert `the alienvault_reputation.ipset` file to a `.cdb` format using the previously downloaded script.

5.- Optional, remove `ipset` file and `python` script.

6.- Assign the right permissions and ownership to the `cdb` file -> WRONG OWNERSHIP IN THE DOCUMENTATION

```
# chown ossec:ossec /var/ossec/etc/lists/blacklist-alienvault
# chmod 660 /var/ossec/etc/lists/blacklist-alienvault
```

It must be

```
# chown wazuh:wazuh /var/ossec/etc/lists/blacklist-alienvault
# chmod 660 /var/ossec/etc/lists/blacklist-alienvault
```

7.- Custom rule to trigger the active response.

8.- Active response settings

Here, we could clarify the line that was added to the ruleset section of the ossec.conf file.

9.- Restart manager

#### Steps to generate alerts

1.- Log into the attacker’s system (the monitored Windows endpoint).

2.- Connect to the victim’s system (the Apache server in the monitored CentOS 8 endpoint) from a web browser.



#### Query the alerts

We can see the active response use case is working properly, the custom rule has detected that the IP was in the AlienVault reputation database (**100100**); and that host was blocked, and unblocked after the timeout (rules **651** and **652**, respectively).

![image](https://user-images.githubusercontent.com/57481331/160594457-a92b4e5b-41bb-4bed-80b9-053f2d101cba.png)


Despite the fact that the screenshot I took looks similar to the documentation page's, I would like to point out 2 considerations:

1.- The alert generated by 652 (host unblocked) is not shown on the documentation page. In my opinion, it should appear as it is part of the POC.
2.- I don't know why there are a lot of alerts generated by 651 and 100100. These alerts were also generated when the screenshot was taken on the Wazuh documentation page. Is this expected?

### CONCLUSION: Blocking a malicious actor POC

In Prerequisites:

- 🟡 Suggestion: Links or guides on how to install Apache web server in the CentOS 8 host 

In Configuration:

- 🔴 Wrong ownership in step number 6: `wazuh:wazuh` vs `ossec:ossec`, being `wazuh:wazuh` the right one
- 🟡 Suggestion: We could clarify which `<list>` option was the one added to the `ruleset` section of the `ossec.conf` (step 8).

In Query the alerts:

- 🟡 Suggestion: Change the screenshot to also show the alert generated by the rule with ID 652 (unblock) as it is part of the POC.
- 🟡 Question: A lot of alerts generated by 651 and 100100, why?

**Comment by mcarmona99** [comment]
## Active response tests

### BLOG: How to integrate YARA with Wazuh :green_circle:

The [How to integrate YARA with Wazuh blog](https://wazuh.com/blog/how-to-integrate-yara-with-wazuh/) is outdated since Wazuh v4.2.0. This is warned at the beginning of the blog with a note.

This blog will still be useful for users trying to integrate Wazuh with YARA in older versions.


For Wazuh versions greater or equal than v4.2.0, the integration of Wazuh with YARA is available in the Wazuh documentation:

[Documentation > User manual > Capabilities > Active response > Use cases > How to integrate Wazuh with YARA](https://documentation.wazuh.com/current/user-manual/capabilities/active-response/ar-use-cases/wazuh-with-yara.html)

### Active response use cases > How to integrate Wazuh with YARA 🟡 

I will test the How to integrate Wazuh with YARA use case using Wazuh v4.3.0.

Using the same environment as in the last POC: 1 manager installed in CentOS7 + 1 CentOS8 endpoint (agent)

#### Wazuh manager configuration

Everything is OK.

#### Wazuh agent configuration

Default YARA installation (with the `root` user):

YARA path (executable file) = `/usr/bin`

Wrong ownership in the following note:

`Note: Make sure that you have jq installed, and that the yara.sh file ownership is root:ossec and the permissions are 750.
`

We should use `root:wazuh` instead


#### Malware detection


In the **Wazuh alerts** section, JSON alerts are shown but the user does not know how they were created. These rules were created by active response, as it used the YARA executable and logged the result. The YARA scan was done after rules 550 or 554, but there were no mention on how to configure FIM for this rules to generate alerts. Mentioning the configuration of FIM and saying that the file monitored is the one scanned by YARA is necessary before showing the Wazuh alerts generated.

Added `<directories realtime="yes">/test</directories>` to the agent's `ossec.conf`.

After adding a file with a string that should be matched by one of the Yara rules (the first one, `deployment`), no active response alerts are generated. 

After debugging, I have seen that the output of `/usr/bin/yara -w -r /root/yara_rules /test/script.sh` is an error:

`/root/yara_rules(1): error: input in flex scanner failed
`

This error is because I used a directory and included some files with each YARA rule. This is wrong as all the yara rules should be in the same file and that is the file given to the `extra_args`.

I used:

```
yara_rules/
	hiddenwasp_deployment
	hiddenwasp_rootkit
	hiddenwasp_trojan
```
	
And everything should be in a single file, called, for instance, `yara_rules`. This should also be explained in the documentation.

After this changes, I could replicate the Wazuh alerts shown the **Wazuh alerts** section:

I added `script.sh` to the /test folder, which is the one monitored.
`script.sh` content:

`http://103.206.123.13:8080/configUpdate.tar.gz
`

As we can see in the following image, both alerts indicating the modification of the file and the YARA detection are generated:

![image](https://user-images.githubusercontent.com/57481331/161072604-2b76c5c1-4e51-4faa-9b63-74a46dcf13e5.png)



The second and thirds alerts are generated by Yara rules when looking for hexadecimal strings. I will not replicate these 2 alerts as how to generate them is explained in the **Wazuh integration with YARA proof of concept**, which is the following thing to test.

### CONCLUSION: Active response use cases > How to integrate Wazuh with YARA

Wazuh agent configuration
- 🔴 Wrong ownership in the note (Note: Make sure that you have jq installed, and that the yara.sh file ownership is root:ossec and the permissions are 750.). We should use `root:wazuh` instead.

Malware detection
- 🔴 We should add a better introduction to this section. Mention that the Yara rules should be inside the same file; and /path/to/rules in the manager configuration should be the **path to the rule file including the rule file itself**. Also mention a basic FIM configuration and explain the flow: FIM -> AR (YARA) -> CUSTOM ALERT FOR YARA

**Comment by mcarmona99** [comment]
## Active response tests

### Proof of Concept guide > Detecting malware using Yara integration 🔴 



#### Configuration

Steps 1 to 2 are OK.

3.- Add the following configuration to `/var/ossec/etc/ossec.conf` at the Wazuh manager.

In this configuration, the Yara path is `/usr/local/bin` and the Yara rules path is `~/yara_rules.yar`. We should talk about the YARA installation as we did in the Yara use of case (the one tested before this comment). We could also add links to it instead of repeating that information. EDIT: step 5 explains how to install YARA so no more information is needed here.

Step 4 is OK.

5.- Compile and install Yara at the monitored CentOS 8 endpoint.

I am repeating this step to replicate the whole POC (as I had my previous YARA installation).

![image](https://user-images.githubusercontent.com/57481331/161078679-a60d761b-356b-4957-a032-9329167f9d23.png)


Step 6 is OK

7.- Download a malware sample (this is a real malware artifact) and run a Yara scan.

Suggestion: It would be great to include here the YARA scan output.

![image](https://user-images.githubusercontent.com/57481331/161079551-4d83189d-1fdd-4699-b55c-5f58cb6cfc1e.png)


8.- Create the `yara.sh` script in `/var/ossec/active-response/bin/` at the monitored CentOS 8 endpoint. This is necessary for the Wazuh-Yara active response scans.

I have repeated this step again as the script is slightly different than the one used in the previous test.

9.- Change `yara.sh` file owner to `root:ossec` and file permissions to 0750.

Wrong ownership, we should use `root:wazuh`.

#### Steps to generate the alerts

Steps 1 to 2 OK.

3.- Optionally, check the results of the Wazuh-Yara scan in `/var/ossec/logs/active-responses.log` at the monitored CentOS 8 endpoint.

Nothing logged in `active-responses.log`.

Therefore, NO alerts created.

I have been investigating the error and the use case worked when changing the yara rules path from `~/yara_rules.yar` to `/root/yara_rules.yar`

The issue was related to permissions and it could be related to the fact that I was using the `root` user in the process. Changing `~/yara_rules` to `/home/user/yara_rules` will solve this problem in all cases (and the example will be more human-readable).

#### Query the alerts


Alerts for the creation of all malwares are generated.

But talking about Yara alerts, only alerts for the webshell malware were created. This is a wrong behavior as when using Yara out of the active response script, the other files also match Yara rules:


![image](https://user-images.githubusercontent.com/57481331/161093419-270c38ea-6eaf-443b-a29f-90fce882ab61.png)

![image](https://user-images.githubusercontent.com/57481331/161093518-3bb6a106-a166-4c8e-9df2-22434456f931.png)
